### PR TITLE
VTCCA-6206 Upload evidence page interface tests

### PR DESCRIPTION
**Pull request by WillBlakeney**

### CODE REVIEW

File: it/controllers/propertyLinking/UploadControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,1417 @@
...
+  // Licence to occupy page text
+  val licenceHeadingText = "Upload your licence to occupy"
+  val licenceHeadingTextWelsh = "Lanlwytho eich trwydded i feddiannu"
+  val licenceHeadingTextAgent = "Upload your client’s licence to occupy"
+  val licenceHeadingTextAgentWelsh = "Lanlwytho trwydded eich cleient i feddiannu"
+
+  // Service charge statement
+  val serviceChargeHeadingText = "Upload your service charge statement"
+  val serviceChargeHeadingTextWelsh = "Lanlwytho eich datganiad tâl gwasanaeth"
+  val serviceChargeHeadingTextAgent = "Upload your service charge statement"
```
Comment (joeyymoore) [inline it/controllers/propertyLinking/UploadControllerISpec.scala:47]: Is there a reason why for Service charge, Stamp duty, Land registry, Water rate and Utility all don't have a client variation of the text, whereas Business rates bill, license and lease do?

File: it/controllers/propertyLinking/UploadControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,1417 @@
...
+  val backToOtherEvidenceHref = "/business-rates-property-linking/my-organisation/claim/property-links/evidence/OTHER/upload"
+  val errorMessageHref = "#newFileButton"
+  val otherEvidenceErrorMessageHref = "#evidenceType"
+  val evidenceTypeRedirectUrl: String => String = (evidenceType: String) => s"/business-rates-property-linking/my-organisation/claim/property-links/evidence/$evidenceType/upload"
+  val checkYourAnswersRedirectUrl = "/business-rates-property-linking/my-organisation/claim/property-links/summary"
+
+  "UploadController show method" should {
+
+    // IP - English
+    "display the correct content in English for an IP that is uploading a business rates bill" which {
```
Comment (joeyymoore) [inline it/controllers/propertyLinking/UploadControllerISpec.scala:191]: might be worth putting the relationship type in the name of these tests

File: it/controllers/propertyLinking/UploadControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,1417 @@
...
+    "display the correct content in Welsh for an Agent that is uploading a lease" which {
+      commonEvidencePageTests(language = Welsh,
+        evidenceType = "LEASE",
+        userIsAgent = true,
+        relationship = "Occupier",
+        backLink = backToEvidenceQuestionHref,
+        fromCya = false)
+    }
+
+    "display the correct content in Welsh for an Agent that is uploading a licence to occupy" which {
```
Comment (joeyymoore) [inline it/controllers/propertyLinking/UploadControllerISpec.scala:530]: this is just a test heading not a test (i think you need to delete the line below)

File: it/controllers/propertyLinking/UploadControllerISpec.scala (hunk 1 above)
Comment (WillBlakeney) [reply it/controllers/propertyLinking/UploadControllerISpec.scala:47]: I'm not sure why that is. I raised [VTCCA-6490](https://jira.tools.tax.service.gov.uk/browse/VTCCA-6490) to fix this. I added different vals at the top to make it easy to update the tests when this change is made

File: it/controllers/propertyLinking/UploadControllerISpec.scala (hunk 3 above)
Comment (WillBlakeney) [reply it/controllers/propertyLinking/UploadControllerISpec.scala:530]: Good spot, I will fix that

File: it/controllers/propertyLinking/UploadControllerISpec.scala (hunk 2 above)
Comment (WillBlakeney) [reply it/controllers/propertyLinking/UploadControllerISpec.scala:191]: On it 👍

**Review by joeyymoore — APPROVED** [review]

**Pull request merged**